### ISSUE TITLE
Kanboard on mysql binlog problem

### ISSUE THREAD

**Issue by cberni** [kanboard/kanboard#4875]
- Kanboard version: 1.2.20
- Database type and version: mysql 8
- PHP version: php 7.4
- OS: ubuntu 20.04 lts server

I have a Ubuntu server with mysql 8 and only kanboard 1.2.20 on the system. Only kanboard database (less than 10mb) in this system.
The binlog files are growing so fast ending with disk space. (see below)

-rw-r-----  1 mysql mysql 101M Aug 23 14:16  binlog.000004
-rw-r-----  1 mysql mysql 102M Aug 23 14:18  binlog.000005
-rw-r-----  1 mysql mysql 101M Aug 23 14:21  binlog.000006
-rw-r-----  1 mysql mysql 101M Aug 23 14:23  binlog.000007
-rw-r-----  1 mysql mysql 101M Aug 23 14:26  binlog.000008
-rw-r-----  1 mysql mysql 101M Aug 23 14:29  binlog.000009
-rw-r-----  1 mysql mysql 101M Aug 23 14:31  binlog.000010
-rw-r-----  1 mysql mysql 101M Aug 23 14:34  binlog.000011
-rw-r-----  1 mysql mysql 102M Aug 23 14:36  binlog.000012
-rw-r-----  1 mysql mysql 101M Aug 23 14:38  binlog.000013
-rw-r-----  1 mysql mysql 101M Aug 23 14:39  binlog.000014
-rw-r-----  1 mysql mysql 102M Aug 23 14:40  binlog.000015
-rw-r-----  1 mysql mysql 101M Aug 23 14:42  binlog.000016
-rw-r-----  1 mysql mysql 102M Aug 23 14:43  binlog.000017
-rw-r-----  1 mysql mysql 102M Aug 23 14:45  binlog.000018
-rw-r-----  1 mysql mysql 102M Aug 23 14:47  binlog.000019
-rw-r-----  1 mysql mysql 102M Aug 23 14:48  binlog.000020
-rw-r-----  1 mysql mysql 102M Aug 23 14:51  binlog.000021
-rw-r-----  1 mysql mysql 101M Aug 23 14:52  binlog.000022
-rw-r-----  1 mysql mysql 103M Aug 23 14:55  binlog.000023
-rw-r-----  1 mysql mysql 103M Aug 23 15:03  binlog.000024

Is it a bug? how could I solve that?

**Comment by fguillot** [comment]
It doesn't seems to be related to Kanboard.

Take a look to the Mysql documentation: https://dev.mysql.com/doc/refman/8.0/en/purge-binary-logs.html

**Comment by cberni** [comment]
I have another mysql 8 database with 500gb size and don't generate it big like that. But is another system.

**Comment by erikvanberkum** [comment]
I am having the same issue with the binlog files since I have installed Kanboard. I have Kanboard running with more PHP applications on a virtual machine. I have to install Kanboard on a single virtual machine to see if the issue persists and if the issue disappears from the current server.

- Application version: 1.2.20
- PHP version: 7.4.25
- PHP SAPI: apache2handler
- HTTP Client: cURL
- OS version: Linux 5.4.0-90-generic 
- Database driver: mysql
- Database version: 8.0.27-0ubuntu0.20.04.1 
- Ubuntu Linux 20.04.3

**Comment by erikvanberkum** [comment]
Didnt have the time to install a virtual machine yet, but disconnected the php app from the mysql database by removing the password. After doing that the growth of binlog files stopped immediately. 

So my questions is now what is the recommended mysql server version, to not run into this issue again. The problems seems to be with Mysql 8.0.27.

**Comment by reduzent** [comment]
I experience also a quickly growing mysql binlog. This is on a shared mysql server that is also used by other applications. However, of a 100MB binlog file, around 98-99MB is data from our Kanboard intsance. The culprit seems to be changes in the sessions tables. The sessions that are updated there are up to several MB in size, so only a handful of updates makes the binlog grow quickly. The rate varies, but sometimes the 100MB binlog files are filled within 8 minutes. 

My setup:
  * Kanban 1.2.20
  * PHP 7.3.31
  * Debian 10.11
  * MariaDB 10.1.48 (on Debian 9)

**Comment by reduzent** [comment]
> So my questions is now what is the recommended mysql server version, to not run into this issue again. The problems seems to be with Mysql 8.0.27.

I do not think it is related to the MySQL version. The problem seems with large sessions being updated often.

**Comment by erikvanberkum** [comment]
So a few users that keep their tabs open will make it grow quickly, you think?

**Comment by reduzent** [comment]
I don't know what triggers an 'update session' statement, so I cannot tell. I just scanned the binlog files with `mysqlbinlog`.

**Comment by reduzent** [comment]
Or more specifically, the size of the data field in the sessions table is sometimes several MB. A little change in the session data means that all of the data needs to be rewritten -> large binlogs.

**Comment by reduzent** [comment]
Also, I do not think it is an installation issue as it is labelled now. We've been running Kanboard for quite  a while, but got aware of the issue only recently. I observe that all reports in this issue mention Kanboard version 1.2.20.

**Comment by erikvanberkum** [comment]
I have reinstalled Kanboard on an Ubuntu 18.0.4 with a PostgreSQL database. I converted the tables from MySQL to Postgres via CSV export import. I used [Dbeaver community edition](https://dbeaver.io/download/) as a tool to do this.  Will monitor again to see if there are any problems.

**Comment by erikvanberkum** [comment]
On a further note, after moving databases, I saw the following @reduzent has narrowed the problem down to the sessions table.
I expect that the following settings in the settings database table are frequently updating the sessions table:

![image](https://user-images.githubusercontent.com/7335299/144351835-9778cdea-e724-4d90-9c21-a2f20b3c0201.png)

Maybe someone can test if the refresh interval has any influence, by increasing the number of seconds.

**Comment by reduzent** [comment]
> Maybe someone can test if the refresh interval has any influence, by increasing the number of seconds.

@erikvanberkum thanks for the pointer. I increased those values to 100 and 600, respectively, and I don't notice a reduction in the amount of binlog data produced per period. I still get a new 100MB file every few minutes.

**Comment by erikvanberkum** [comment]
Some further reading:

In earlier MySQL versions, binary logging was disabled by default, and was enabled if you specified the --log-bin option. From MySQL 8.0, binary logging is enabled by default, whether or not you specify the --log-bin option. The exception is if you use mysqld to initialize the data directory manually by invoking it with the --initialize or --initialize-insecure option, when binary logging is disabled by default. It is possible to enable binary logging in this case by specifying the --log-bin option. When binary logging is enabled, the log_bin system variable, which shows the status of binary logging on the server, is set to ON.

To disable binary logging, you can specify the --skip-log-bin or --disable-log-bin option at startup. If either of these options is specified and --log-bin is also specified, the option specified later takes precedence. When binary logging is disabled, the log_bin system variable is set to OFF.

Source https://dev.mysql.com/doc/refman/8.0/en/replication-options-binary-log.html

**Comment by reduzent** [comment]
We use master-slave replication, so we need binary logs. However, if this problem is going to persist, we might have to migrate the kanboard database to a dedicated mysql/mariadb instance without binary logs.

**Comment by reduzent** [comment]
Can someone please change the label? This doesn't seem to be an installation issue.

**Comment by reduzent** [comment]
We migrated the database away from the MariaDB cluster that uses replication and thus needs binlogs to a local PostgreSQL database instance.
